crates/uv-python/src/discovery.rs

```diff
@@ -1381,6 +1393,18 @@ impl PythonRequest {
         }
     }
 
+    pub(crate) fn allows_alternative_implementations(&self) -> bool {
+        match self {
+            Self::Default => false,
+            Self::Any => true,
```

Comment:
Why is `Any` true but (e.g.) `Version` and `Default` false?

Reply:
Per #7526 we want to distinguish between discovery that allows any interpreter and discovery of a good default interpreter. `Any` should never filter an interpreter. `Default` filters alternative implementations because they require opt-in. `Version` filters alternative implementations because asking for `3.13` isn't enough to select PyPy, we need `pypy@3.13` which is `ImplementationVersion`.